packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -0,0 +1,213 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ReactElement, ReactNode} from 'react';
+import { useRef, useState } from 'react';
+
+import BpkBreakpoint, { BREAKPOINTS } from '../../bpk-component-breakpoint';
+import BpkSelectableChip, { BpkDismissibleChip, BpkIconChip, BpkDropdownChip, CHIP_TYPES } from '../../bpk-component-chip';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import FilterIconSm from '../../bpk-component-icon/sm/filter';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkMobileScrollContainer from '../../bpk-component-mobile-scroll-container';
+import { cssModules } from '../../bpk-react-utils';
+
+import Nudger from './Nudger';
+
+import STYLES from './BpkChipGroup.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export const CHIP_GROUP_TYPES = {
+  rail: 'rail',
+  wrap: 'wrap',
+};
+
+export const CHIP_COMPONENT = {
+  selectable: 'selectable',
+  dismissible: 'dismissible',
+  dropdown: 'dropdown',
+  icon: 'icon',
+};
+
+const CHIP_COMPONENT_MAP = {
+  [CHIP_COMPONENT.selectable]: BpkSelectableChip,
+  [CHIP_COMPONENT.dismissible]: BpkDismissibleChip,
+  [CHIP_COMPONENT.dropdown]: BpkDropdownChip,
+  [CHIP_COMPONENT.icon]: BpkIconChip,
+}
+
+export type ChipGroupType = (typeof CHIP_GROUP_TYPES)[keyof typeof CHIP_GROUP_TYPES];
+export type ChipStyleType = (typeof CHIP_TYPES)[keyof typeof CHIP_TYPES];
+export type ChipComponentType = (typeof CHIP_COMPONENT)[keyof typeof CHIP_COMPONENT];
+
+export type SingleSelectChipItem = {
+  text: string;
+  accessibilityLabel?: string;
+  leadingAccessoryView?: ReactNode;
+  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
+};
+
+export type ChipItem = {
+  component?: ChipComponentType;
+  onClick?: (selected: boolean, index: number) => void;
+  selected?: boolean;
+  hidden?: boolean;
+} & SingleSelectChipItem;
+
+export type CommonProps = {
+  ariaLabel?: string;
+  type?: ChipGroupType;
+  chipStyle?: ChipStyleType;
+  // only required when type = rail
+  leadingNudgerLabel?: string;
+  trailingNudgerLabel?: string;
```

Comment:
This doesn't feel ideal as it can lead to these props not being populated easily. Either:
1. These props are required, but they are not needed for `wrap` chip groups which will be a little annoying for consumers.
2. Add a warning console log when type = `rail` and these are empty - see this pattern in other components but still no guarantee they'll be populated.
3. Refactor the interface and make "conditional" props [like this](https://dev.to/maissenayed/conditional-react-props-with-typescript-43lg). This ensures props are set correctly but at the const of a more complicated interface (and being tricky to implement).